core/src/main/java/com/microsoft/applicationinsights/internal/jmx/JmxDataFetcher.java

```diff
@@ -46,6 +46,7 @@
 public class JmxDataFetcher {
 
     private static final Logger logger = LoggerFactory.getLogger(JmxDataFetcher.class);
+    private static ArrayList<JmxAttributeData> warningShown= new ArrayList<>();
```

Comment:
set will perform `contains` better, and concurrent data structure is important since accessed across multiple threads
```suggestion
    private static Set<JmxAttributeData> warningShown= Collections.newSetFromMap(new ConcurrentHashMap<>());
```